### PR TITLE
Add support for running shell with alternative credentials

### LINKED ISSUE #23: resin-git needs to spawn shell command under alternative uid/gid

### PR DESCRIPTION
**Pull request by wrboyce**
Fixes #23
Change-Tye: minor

### CODE REVIEW

File: resin/main.go [hunk 1]
```diff
@@ -124,6 +133,16 @@ func main() {
 		fixPathCheckExists("auth-failed-banner")
 	}
 
+	shellCreds := &syscall.Credential{}
+	if viper.IsSet("shell-uid") || viper.IsSet("shell-gid") {
```
Comment (josephroberts) [inline resin/main.go]: Do you need to check if the value is set? Can you just check if the value is greater than 0 instead as the default value is -1?

Comment (wrboyce) [inline resin/main.go]: Fair point, fixed!

File: sshproxy.go [hunk 2]
```diff
@@ -307,6 +309,10 @@ func (s *Server) launchCommand(channel ssh.Channel, cmd *exec.Cmd, terminal *pty
 		}
 	}
 
+	if s.shellCreds != nil {
+		cmd.SysProcAttr = &syscall.SysProcAttr{Credential: s.shellCreds}
```
Comment (josephroberts) [inline sshproxy.go]: I am not sure I totally understand the issue... how does this line fix it?

Comment (wrboyce) [inline sshproxy.go]: `resin-git` needs to spawn the shell process as a different user, setting `SysProcAttr.Credential` was the only way I could see of achieving this without messing about using `sudo`.

Comment (josephroberts) [inline sshproxy.go]: OK cool, looks like a good approach to me

**Pull request merged**